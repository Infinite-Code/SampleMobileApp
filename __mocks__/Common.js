import React from 'react';
import {
    View,
} from 'react-native';

export class EmptyView extends React.Component {
    render() {
        return <View></View>;
    }
}
