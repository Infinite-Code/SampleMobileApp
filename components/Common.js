import React from 'react';
import {
    Text,
} from 'react-native';


export class Title extends React.Component {
    render() {
        return (
            <Text style={{marginTop: 20, fontSize: 20, textAlign: 'center'}}>
                {this.props.children}
            </Text>
        );
    }
}
