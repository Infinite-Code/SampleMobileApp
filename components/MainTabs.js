/**
 * @flow
 */

import React from 'react';
import {
  Button,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import {
  createNavigator,
  createNavigationContainer,
  TabRouter,
  addNavigationHelpers,
} from 'react-navigation';

const MyNavScreen = ({ navigation, banner }) => (
  <ScrollView>
    <Text>{banner}</Text>
    <Button
      onPress={() => {
        navigation.goBack(null);
      }}
      title="Go back"
    />
  </ScrollView>
);

const HomeScreen = ({ navigation }) => (
  <MyNavScreen banner="Home" navigation={navigation} />
);

const FavScreen = ({ navigation }) => (
  <MyNavScreen banner="Favourites" navigation={navigation} />
);

const AddScreen = ({ navigation }) => (
  <ScrollView>
    <Text>Choose the type of post you want to create</Text>
    <Button onPress={() => {}} title="Announcements, Shout Out" />
    <Button onPress={() => {}} title="Open Question" />
    <Button onPress={() => {}} title="Quick Poll" />
    <Button onPress={() => {}} title="Celebration, Gratitude" />
  </ScrollView>
);

const SearchScreen = ({ navigation }) => (
  <MyNavScreen banner="Search" navigation={navigation} />
);

const ProfileScreen = ({ navigation }) => (
  <MyNavScreen banner="Profile" navigation={navigation} />
);

const CustomTabBar = ({ navigation }) => {
  const { routes } = navigation.state;
  return (
    <View style={styles.tabContainer}>
      {routes.map(route => (
        <TouchableOpacity
          onPress={() => navigation.navigate(route.routeName)}
          style={styles.tab}
          key={route.routeName}
        >
          <Text>{route.routeName}</Text>
        </TouchableOpacity>
      ))}
    </View>
  );
};

const CustomTabView = ({ router, navigation }) => {
  const { routes, index } = navigation.state;
  const ActiveScreen = router.getComponentForState(navigation.state);
  return (
    <View style={styles.container}>
      <ActiveScreen
        style={{flex: 1}}
        navigation={addNavigationHelpers({
          ...navigation,
          state: routes[index],
        })}
      />
      <CustomTabBar navigation={navigation} />
    </View>
  );
};

const CustomTabRouter = TabRouter(
  {
    Home: {
      screen: HomeScreen,
      path: 'home',
    },
    Favourites: {
      screen: FavScreen,
      path: 'fav',
    },
    Add: {
      screen: AddScreen,
      path: 'settings',
    },
    Search: {
      screen: SearchScreen,
      path: 'search',
    },
    Profile: {
      screen: ProfileScreen,
      path: 'profile',
    },
  },
  {
    // Change this to start on a different tab
    initialRouteName: 'Home',
  }
);

const MainTabs = createNavigationContainer(
  createNavigator(CustomTabRouter)(CustomTabView)
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
    flexDirection: 'column',
    marginTop: Platform.OS === 'ios' ? 20 : 0,
  },
  tabContainer: {
    flexDirection: 'row',
    height: 48,
  },
  tab: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    margin: 4,
    borderWidth: 1,
    borderColor: '#ddd',
    borderRadius: 4,
  },
});

export default MainTabs;
