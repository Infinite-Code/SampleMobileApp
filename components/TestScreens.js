import React from 'react';
import {
    Button,
    View,
} from 'react-native';
import PropTypes from 'prop-types'

import { Title } from './Common';


export class HomeScreen extends React.Component {
    render() {
        const navigate = this.props.navigation.navigate;
        return (
            <View>
                <Title>Home</Title>

                <Button
                    onPress={() => navigate('DrawerOpen')}
                    title="Open Drawer" />

                <Title>{this.props.userName}, {this.props.email}</Title>
                <Button
                    onPress={this.props.onUpdateUserName}
                    title="Update Username" />

                <Title>{this.props.timestamp}</Title>
                <Button
                    onPress={this.props.onUpdateTimestamp}
                    title="Update Timestamp" />

                <Title>{this.props.description}</Title>
                <Button
                    onPress={this.props.onGetDescription}
                    title="Get @cheeming bio" />
            </View>

        );
    }
}
HomeScreen.propTypes = {
    userName: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    timestamp: PropTypes.string.isRequired,
    onUpdateUserName: PropTypes.func.isRequired,
    onUpdateTimestamp: PropTypes.func.isRequired,
    onGetDescription: PropTypes.func.isRequired,
}


export class ProfileScreen extends React.Component {
    render() {
        return (
            <View>
                <Title>Profile</Title>
            </View>
        );
    }
}

export class AboutScreen extends React.Component {
    render() {
        const navigation = this.props.navigation;
        return (
            <View>
                <Title>About</Title>
                <Button
                    onPress={() => navigation.goBack()}
                    title="Go Back" />
            </View>
        );
    }
}

export default {
    HomeScreen,
    ProfileScreen,
    AboutScreen,
};
