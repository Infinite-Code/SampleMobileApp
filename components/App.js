import {
    TabNavigator,
    DrawerNavigator,
} from 'react-navigation';

import TabBarBottom from './TabBarBottom';
import TestScreens from './TestScreens';


const MainTab = TabNavigator({
    Home: {screen: TestScreens.HomeScreen},
    Profile: {screen: TestScreens.ProfileScreen},
}, {
    tabBarComponent: TabBarBottom,
    tabBarPosition: 'bottom',
});

const App = DrawerNavigator({
    Home: {screen: MainTab},
    About: {screen: TestScreens.AboutScreen},
});
export default App;
