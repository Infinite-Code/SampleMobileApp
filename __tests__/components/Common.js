import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { Title } from '../../components/Common';

it('Title.render', () => {
    const tree = renderer.create(
        <Title />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
