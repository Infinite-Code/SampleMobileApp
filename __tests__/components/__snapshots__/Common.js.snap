// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title.render 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "fontSize": 20,
      "marginTop": 20,
      "textAlign": "center",
    }
  }
/>
`;
